library/src/camera_impl.cxx

```diff
@@ -33,22 +33,22 @@ camera& camera_impl::setPosition(const vector3_t& pos)
 }
 
 //----------------------------------------------------------------------------
-camera::vector3_t camera_impl::getPosition()
+point3_t camera_impl::getPosition()
 {
-  vector3_t pos;
+  point3_t pos;
   this->getPosition(pos);
   return pos;
 }
 
 //----------------------------------------------------------------------------
-void camera_impl::getPosition(vector3_t& pos)
+void camera_impl::getPosition(point3_t& pos)
 {
   vtkCamera* cam = this->GetVTKCamera();
   cam->GetPosition(pos.data());
 }
 
 //----------------------------------------------------------------------------
-camera& camera_impl::setFocalPoint(const vector3_t& foc)
+camera& camera_impl::setFocalPoint(const point3_t& foc)
```

Comment:
Can you use initializer list here with the current implementation? I'm expecting user wanting to call `setFocalPoint({ 0, 0, 0})`

Reply:
should I put that in types.h or in camera_impl::setFocalPoint ?

Reply:
Actually, Maybe just add setters with 3 doubles.

Reply:
I do think that making sure that `std::array<double, 3>`'s initializer list constructor is visible would be useful. The downside of setters with 3 doubles is the loss of typing.

Reply:
> Actually, Maybe just add setters with 3 doubles.

I dont think this is the right way to go, why even defining a vector3_t in that case ?

> that std::array<double, 3>'s initializer list constructor is visible would be useful. 

I also think so